### ISSUE TITLE
Potential PR for lifecycle callbacks on junction tables (used with through-associations)

### ISSUE THREAD

**Issue by devinivy** [balderdashy/waterline#707]
I'm using through associations in waterline, and it seems to be working alright after some hacking, but I can't use lifecycle callbacks.  I'd very much like to!  The issue is that junction tables are re-loaded as collections.  So, my lifecycle callbacks register the first time and are overwritten the second time.  See here: https://github.com/balderdashy/waterline/blob/f8b499ef549f864a813b1eca5bb591f0ec210ac6/lib/waterline.js#L127-L138

If I issue a pull request with relevant tests, would you consider accepting this mini-"feature"?  I would approach this problem by specifically disallowing junction tables to overwrite existing collections.  I would key them on collection identity.

**Comment by HaNdTriX** [comment]
+1

**Comment by sailsbot** [comment]
Thanks for posting, @devinivy.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
